var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -92,14 +93,15 @@ class Gdal(AutotoolsPackage):
     # Required dependencies
     depends_on('libtiff@3.6.0:')  # 3.9.0+ needed to pass testsuite
     depends_on('libgeotiff@1.2.1:')
-    depends_on('json-c')
+    depends_on('json-c', when='@2:')
```

Comment:
I would drop the `when` from this line. Otherwise you'll find that `spack spec gdal ^json-c@1.2.3` crashes.

Reply:
I am not sure why, but this works for me. I left it in since I had to change the version ranges (see later comment).